packages/block-library/src/spacer/edit.js

```diff
@@ -51,50 +61,122 @@ const SpacerEdit = ( {
 		setIsResizing( false );
 	};
 
-	return (
-		<>
-			<View { ...useBlockProps() }>
+	const handleOnHorizontalResizeStop = ( event, direction, elt, delta ) => {
+		onResizeStop();
+		const spacerWidth = Math.min(
+			parseInt( width + delta.width, 10 ),
+			MAX_SPACER_WIDTH
+		);
+		updateWidth( spacerWidth );
+		setIsResizing( false );
+	};
+
+	const resizableBoxWithOrientation = ( blockOrientation ) => {
+		if ( blockOrientation === 'horizontal' ) {
+			return (
 				<ResizableBox
 					className={ classnames(
 						'block-library-spacer__resize-container',
+						'resize-horizontal',
 						{
 							'is-selected': isSelected,
 						}
 					) }
 					size={ {
-						height,
+						width,
+						height: 32,
 					} }
-					minHeight={ MIN_SPACER_HEIGHT }
+					minWidth={ MIN_SPACER_WIDTH }
 					enable={ {
 						top: false,
-						right: false,
-						bottom: true,
+						right: true,
+						bottom: false,
 						left: false,
 						topRight: false,
 						bottomRight: false,
 						bottomLeft: false,
 						topLeft: false,
 					} }
 					onResizeStart={ handleOnResizeStart }
-					onResizeStop={ handleOnResizeStop }
+					onResizeStop={ handleOnHorizontalResizeStop }
 					showHandle={ isSelected }
 					__experimentalShowTooltip={ true }
 					__experimentalTooltipProps={ {
-						axis: 'y',
-						position: 'bottom',
+						axis: 'x',
+						position: 'corner',
 						isVisible: isResizing,
 					} }
 				/>
+			);
+		}
+
+		return (
+			<ResizableBox
+				className={ classnames(
+					'block-library-spacer__resize-container',
+					{
+						'is-selected': isSelected,
+					}
+				) }
+				size={ {
+					height,
+				} }
+				minHeight={ MIN_SPACER_HEIGHT }
+				enable={ {
+					top: false,
+					right: false,
+					bottom: true,
+					left: false,
+					topRight: false,
+					bottomRight: false,
+					bottomLeft: false,
+					topLeft: false,
+				} }
+				onResizeStart={ handleOnResizeStart }
+				onResizeStop={ handleOnVerticalResizeStop }
+				showHandle={ isSelected }
+				__experimentalShowTooltip={ true }
+				__experimentalTooltipProps={ {
+					axis: 'y',
+					position: 'bottom',
+					isVisible: isResizing,
+				} }
+			/>
+		);
+	};
+
+	useEffect( () => {
+		if ( orientation === 'horizontal' && ! width ) {
+			updateWidth( 72 );
+			updateHeight( 0 );
+		}
+	}, [] );
```

Comment:
This bit is the only one I don't understand, everything else makes perfect sense in the code.

Reply:
Looks like this is being used as the default horizontal spacer size? Probably to get around the fact that we only want these defaults in the vertical or horizontal position, not at the same time.
```
"attributes": {
		"height": {			
			"type": "number",			
			"default": 100				
		},
		"width": {
			"type": "number"
		}			}
```

Would it be possible to move the magic numbers to a named constant?

Reply:
If the spacer is horizontal, I have to set the height to 0 because otherwise it'll stay at the default 100px, which will likely break whatever layout it's inserted in. The default height has to exist because most spacer blocks won't get orientation context so need that value to start with.

> Would it be possible to move the magic numbers to a named constant?

We don't really use named constants for default attribute values, as the values only occur in the one place. I think named constants are useful where we need values to match in multiple places, but otherwise they just make the code harder to read 😅